src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -390,15 +402,38 @@ bool System.Collections.IList.Contains(Object item)
         // whichever is larger.
         private void EnsureCapacity(int min) {
             if (_items.Length < min) {
-                int newCapacity = _items.Length == 0? _defaultCapacity : _items.Length * 2;
+                int newCapacity = _items.Length == 0 ? _defaultCapacity : _items.Length * 2;
                 // Allow the list to grow to maximum possible capacity (~2G elements) before encountering overflow.
                 // Note that this check works even when _items.Length overflowed thanks to the (uint) cast
                 if ((uint)newCapacity > Array.MaxArrayLength) newCapacity = Array.MaxArrayLength;
                 if (newCapacity < min) newCapacity = min;
                 Capacity = newCapacity;
             }
         }
-   
+
+        [MethodImpl(MethodImplAttributes.NoInlining)]
+        private void IncreaseCapacity()
+        {
+            int length = _items.Length;
+            if (_size == Array.MaxArrayLength)
+            {
+                // New length would be too large for array
+                ThrowHelper.ThrowLengthArgumentOutOfRange_ArgumentOutOfRange_NeedNonNegNum();
```

Comment:
Couldn't find docs saying any exception could be thrown; it would call `set_Capacity` with a negative number, which would call the array .ctor with negative number ` new T[value]`. 

Should throw OutOfMemoryExecption instead?